addons/mail/static/src/models/composer/composer.js

```diff
@@ -824,7 +824,34 @@ function factory(dependencies) {
          * @param {string} mentionKeyword
          */
         async _updateSuggestedPartners(mentionKeyword) {
-            const mentions = await this.async(() => this.env.services.rpc(
+            if (this.thread) {
+                const localMentions = this.thread.searchLocalSuggestedPartners(mentionKeyword);
+                if (localMentions.length) {
+                    const extraMentions = localMentions.filter(partner => !partner.user);
+                    const mainMentions = localMentions.filter(partner => partner.user);
+                    this.update({
+                        extraSuggestedPartners: !extraMentions ? [['unlink-all']] : [[
+                            'replace',
+                            localMentions.filter(partner => !partner.user),
+                        ]],
+                        mainSuggestedPartners: !mainMentions ? [['unlink-all']] : [[
+                            'replace',
+                            localMentions.filter(partner => partner.user),
+                        ]],
+                    });
+                    const activeSuggestedPartner = mainMentions.length > 0 ? mainMentions[0] : extraMentions[0];
+                    if (!activeSuggestedPartner) {
+                        return;
+                    }
+                    this.update({
+                        activeSuggestedPartner: [['link', activeSuggestedPartner]],
+                        hasToScrollToActiveSuggestion: true,
+                    });
+                    return;
```

Comment:
Not a full review yet, but since I saw the PR, I feel this is too much duplicate with the process we do in the other case too. Maybe it would be best if we had a new method to set mention lists given in param and doing this job, independently of the list's origin.

Reply:
Used a function like you said.
Really similar to `_setFirstActiveSuggestion` (or something like that) but you have to unlink if none found and set the scroll attr so using some params to that function would make it hard to understand I think... So I prefered using another function :)